ramls/inventory.raml

```diff
@@ -76,7 +77,30 @@ resourceTypes:
               body:
                 text/plain:
                   example: "Internal server error"
-
+    /move:
+      type:
+        collection-item:
+          exampleItem: !include examples/items_move.json
+          schema: move
+      post:
+        responses:
+          201:
+            description: "Items moved to another holdings record"
+          404:
+            description: "Some of items not moved"
```

Comment:
Yes, you are correct. Do you agree with this?

Reply:
I don't think that a 404 Not Found should be used to report that some items were not moved.

I don't understand how those fit together.

I think it depends why an item cannot be moved. Personally, I think if this occurs after the initial validation and some items can / have been moved, it would be clearer to return a 200 which describes which items have been moved and which ones have not (and why).

Reply:
As an option we can return 202 Accepted instead of 201 Created just to indicate that request was accepted but part of items couldn't updated and completely remove 404 from contract. But from my perspective we need to show that smth goes wrong by http status. 

@NikitaSedyx from the UI perspective what option is more appropriate to show error: 1) return 404 with list of non-updated item ids or 2) return all the time successful http code (even though some of items weren't updated) with list of non-updated item ids if some of them weren't updated?

Reply:
> As an option we can return 202 Accepted instead of 201 Created just to indicate that request was accepted but part of items couldn't updated and completely remove 404 from contract. But from my perspective we need to show that smth goes wrong by http status.

I agree that 404 is not appropriate and should be removed. I also agree that the response needs to convey that something went wrong. 

I don't think 202 is an appropriate status code to convey partial success or failure, that is both synchronous and known at the point of initial response.

As I understand it, 202 is intended for asynchronous processing of a request (and is meant to also convey a way of determining the status of that processing).

`The 202 (Accepted) status code indicates that the request has been accepted for processing, but the processing has not been completed.` (https://tools.ietf.org/html/rfc7231#section-6.3.3)

I'm not sure there is a status code that can convey that partial status appropriately on it's own. 

As I understand, HTTP is intended to be resource-centric. This can present challenges when a resource represents a process, especially when that process involves manipulating more than one resources. 

I think the scenario being described, from the perspective of the resource describing the process, has been handled ok. That within that process, some parts have succeeded and others have failed is below that level and cannot be conveyed in a single status.

Alternatively, any single failure could be considered a failure of the process overall, if that works better for folks. Either way, the client is going to need to know that this status really means that it needs to `investigate further by inspecting the body (or other state of the system)`. 

Personally I think that 200 is more suited to this investigation than 4XX or 5XX.

I'm sorry that I didn't manage to convey this challenge better in the discussions on the design document when I suggested that it might be better for a client to submit multiple requests rather than a group / batch.

Reply:
API will return 201 all the time. In case of some failures it will return list of nonUpdatedIds but the same http status 201.

Reply:
@khandramai I personally think that responding with an overall process status is a good idea.

Please can you help me understand why 201 CREATED was chosen over 200 OK? What resource is being created during this process?

Reply:
I mean some kind we "virtual move" created. It looks not like pure REST but some action was performed and we have result of items moving. But I'm fine with 200 if you think that it is more appropriate.

Reply:
Personally, I think 200 is more appropriate, as the process completed and no new resources were created.

This might be because I'm missing what this `virtual move` is. Does it change any other state of the system other than the items that have been moved?

Reply:
No, just item's state will be updated.

Reply:
OK, I created 200 instead of 201.

Reply:
Thank you